percona-xtradb-cluster-5.7-backup/backup.sh

```diff
@@ -187,9 +187,70 @@ function backup_s3() {
     echo '[INFO] Backup was finished successfully'
 }
 
+azure_auth_header_file() {
+	hex_tmp=$(mktemp)
+	signature_tmp=$(mktemp)
+	auth_header_tmp=$(mktemp)
+
+	params="$1"
+	request_date="$2"
+
+	echo -n "$AZURE_ACCESS_KEY" | base64 -d -w0 | hexdump -ve '1/1 "%02x"' >"$hex_tmp"
+	headers="x-ms-date:$request_date\nx-ms-version:2021-06-08"
+	resource="/$AZURE_STORAGE_ACCOUNT/$AZURE_CONTAINER_NAME"
+	string_to_sign="GET\n\n\n\n\n\n\n\n\n\n\n\n${headers}\n${resource}\n${params}"
+	printf '%s' "$string_to_sign" | openssl dgst -sha256 -mac HMAC -macopt "hexkey:$(cat "$hex_tmp")" -binary | base64 -w0 >"$signature_tmp"
+	echo -n "Authorization: SharedKey $AZURE_STORAGE_ACCOUNT:$(cat "$signature_tmp")" >"$auth_header_tmp"
+	echo "$auth_header_tmp"
+}
+
+is_object_exist_azure() {
+	object="$1"
+	connection_string="$ENDPOINT/$AZURE_CONTAINER_NAME?comp=list&restype=container"
+	request_date=$(LC_ALL=en_US.utf8 TZ=GMT date "+%a, %d %h %Y %H:%M:%S %Z")
+	header_version="x-ms-version: 2021-06-08"
+	header_date="x-ms-date: $request_date"
+	header_auth_file=$(azure_auth_header_file "comp:list\nrestype:container" "$request_date")
+
+	res=$(curl -s -H "$header_version" -H "$header_date" -H "@$header_auth_file" ${connection_string} | grep $object)
+
+	if [[ ${#res} -ne 0 ]]; then
+		return 1
+	fi
+}
+
+function backup_azure() {
```

Comment:
why `function` here and not in other places? I recommend deleting it: https://github.com/koalaman/shellcheck/wiki/SC2112

Reply:
https://github.com/percona/percona-docker/pull/623/commits/35e5d5a64e6f84a5934fd43c7da3e77e5811b8a6